src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -322,130 +322,143 @@ public IActionResult GetHistory([FromQuery] WalletHistoryRequest request)
                 WalletHistoryModel model = new WalletHistoryModel();
 
                 // Get a list of all the transactions, with the addresses associated with them.
-                List<FlatHistory> items = this.walletManager.GetHistory(request.WalletName).ToList().OrderByDescending(o => o.Transaction.CreationTime).Take(200).ToList();
+                IEnumerable<AccountHistory> accountsHistory = this.walletManager.GetHistory(request.WalletName, request.AccountName);
 
-                // Represents a sublist containing only the transactions that have already been spent.
-                List<FlatHistory> spendingDetails = items.Where(t => t.Transaction.SpendingDetails != null).ToList();
+                foreach (var accountHistory in accountsHistory)
+                {
+                    List<TransactionItemModel> transactionItems = new List<TransactionItemModel>();
 
-                // Represents a sublist of transactions associated with receive addresses + a sublist of already spent transactions associated with change addresses.
-                // In effect, we filter out 'change' transactions that are not spent, as we don't want to show these in the history.
-                List<FlatHistory> history = items.Where(t => !t.Address.IsChangeAddress() || (t.Address.IsChangeAddress() && !t.Transaction.IsSpendable())).ToList();
+                    List<FlatHistory> items = accountHistory.History.OrderByDescending(o => o.Transaction.CreationTime).Take(200).ToList();
```

Comment:
That's just because we want to limit the number of processing we do and 200 history items seemed like enough info for the user. We'll change that when we have pagination ready.